File: api/Models/Product.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    [Index(nameof(Barcode), IsUnique = true)]
    public class Product
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = default!;
        [Required]
        [MaxLength(30)]
        public string Barcode { get; set; } = default!;
        [Required]
        [MaxLength(150)]
        public string Description { get; set; } = default!;
        public int PresentationId { get; set; }
        public Presentation Presentation { get; set; } = default!;
        public ICollection<StoreProduct> StoreProducts { get; set; } = new List<StoreProduct>();
        public ICollection<BuyOrderDetail> BuyOrderDetails { get; set; } = new List<BuyOrderDetail>();
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: api/app/models/product.ts
import { DateTime } from 'luxon';
import { BaseModel, belongsTo, column, hasMany } from '@adonisjs/lucid/orm';
import BuyOrderDetail from './buy_order_detail.js';
import Presentation from './presentation.js';
import type { BelongsTo, HasMany } from '@adonisjs/lucid/types/relations';
import StoreProduct from './store_product.js';
import SaleDetail from './sale_detail.js';

export default class Product extends BaseModel {
  @column({ isPrimary: true })
  declare id: number;

  @column()
  declare name: string;

  @column()
  declare barcode: string;

  @column()
  declare description: string;

  @column()
  declare presentationId: number;

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime;

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime;

  @column.dateTime()
  declare deletedAt: DateTime | null;

  @belongsTo(() => Presentation)
  declare presentation: BelongsTo<typeof Presentation>;

  @hasMany(() => BuyOrderDetail)
  declare buyOrderDetails: HasMany<typeof BuyOrderDetail>;

  @hasMany(() => StoreProduct)
  declare storeProducts: HasMany<typeof StoreProduct>;

  @hasMany(() => SaleDetail)
  declare saleDetails: HasMany<typeof SaleDetail>;
}
